grid2op/tests/test_ma.py

```diff
@@ -597,6 +608,311 @@ def check_mask_topo_vect(self, ma_env, domain, space="action"):
+                type_action = 'line_ex_set_bus',
+                is_set = True,
+                value = 2
+            )
+            
+            
+            # Test for storages
+            self.check_local2global(
+                agent = agent,
+                n = self.ma_env._subgrids_cls['action'][agent].n_storage,
+                orig_ids = self.ma_env._subgrids_cls['action'][agent].storage_orig_ids,
+                type_action = 'storage_set_bus',
+                is_set = True,
+                value = 2
+            )
+
+
+    def test_local_action_to_global_change_bus(self):
+        # The global env must have at least one storage
+        assert self.ma_env._cent_env.n_storage > 0
+        
+        for agent in self.ma_env.agents:
+            
+            # Test for loads
+            self.check_local2global(
+                agent = agent,
+                n = self.ma_env._subgrids_cls['action'][agent].n_load,
+                orig_ids = self.ma_env._subgrids_cls['action'][agent].load_orig_ids,
+                type_action = 'load_change_bus',
+                is_set = False,
+                value = 2
+            )
+
+            
+            # Test for gens
+            self.check_local2global(
+                agent = agent,
+                n = self.ma_env._subgrids_cls['action'][agent].n_gen,
+                orig_ids = self.ma_env._subgrids_cls['action'][agent].gen_orig_ids,
+                type_action = 'gen_change_bus',
+                is_set = False,
+                value = 2
+            )
+            
+            # Test for line_ors
+            self.check_local2global(
+                agent = agent,
+                n = self.ma_env._subgrids_cls['action'][agent].n_line,
+                orig_ids = self.ma_env._subgrids_cls['action'][agent].line_orig_ids,
+                type_action = 'line_or_change_bus',
+                is_set = False,
+                value = 2
+            )
+            
+            # Test for line_exs
+            self.check_local2global(
+                agent = agent,
+                n = self.ma_env._subgrids_cls['action'][agent].n_line,
+                orig_ids = self.ma_env._subgrids_cls['action'][agent].line_orig_ids,
+                type_action = 'line_ex_change_bus',
+                is_set = False,
+                value = 2
+            )
+            
+            
+            # Test for storages
+            self.check_local2global(
+                agent = agent,
+                n = self.ma_env._subgrids_cls['action'][agent].n_storage,
+                orig_ids = self.ma_env._subgrids_cls['action'][agent].storage_orig_ids,
+                type_action = 'storage_change_bus',
+                is_set = False,
+                value = 2
+            )
+        
+    
+    def test_local_action_to_global_change_line_status(self):
+        # TODO comments
+        for agent in self.ma_env.agents:
+            
```

Comment:
??? TODO comments ...

I should just close the PR and let you start again

Reply:
done